test/cloudant.test.js

```diff
@@ -324,6 +324,26 @@ describe('cloudant constructor', function() {
       result.url.should.equal(ds.settings.url);
     });
   });
+  it('should convert first part of url path to database name', function() {
+    var ds = getDataSource();
+    ds.settings = _.clone(ds.settings) || {};
+    var result = {};
+    ds.settings.Driver = function(options) {
+      result = options;
+    };
+    ds.settings.url = 'https://fakeuser:fakepass@definitelynotreal.cloudant.com:8080/some/random/path';
+    var connector = Cloudant.initialize(ds, function(err) {
+      should.exist(err);
+      should.exist(result.url);
+      result.url.should.equal('https://fakeuser:fakepass@definitelynotreal.cloudant.com:8080');
+      result.database.should.not.equal('some');
+    });
+    ds.settings.database = undefined;
```

Comment:
If `Cloudant.initialize()` is a async function, I think line#341-line#345 should be moved into line#335's callback function?